File: Roadmap/00 - SINTAXIS, VARIABLES, TIPOS DE DATOS Y HOLA MUNDO/go/n0hagonada.go
package main

import "fmt"

/*
 * ¿Preparad@ para aprender o repasar el lenguaje de programación que tú quieras?
 * - Recuerda que todas las instrucciones de participación están en el
 *   repositorio de GitHub.
 *
 * Lo primero... ¿Ya has elegido un lenguaje?
 * - No todos son iguales, pero sus fundamentos suelen ser comunes.
 * - Este primer reto te servirá para familiarizarte con la forma de participar
 *   enviando tus propias soluciones.
 *
 * EJERCICIO:
 * - Crea un comentario en el código y coloca la URL del sitio web oficial del
 *   lenguaje de programación que has seleccionado.
 * - Representa las diferentes sintaxis que existen de crear comentarios
 *   en el lenguaje (en una línea, varias...).
 * - Crea una variable (y una constante si el lenguaje lo soporta).
 * - Crea variables representando todos los tipos de datos primitivos
 *   del lenguaje (cadenas de texto, enteros, booleanos...).
 * - Imprime por terminal el texto: "¡Hola, [y el nombre de tu lenguaje]!"
 *
 * ¿Fácil? No te preocupes, recuerda que esta es una ruta de estudio y
 * debemos comenzar por el principio.
 */
// https://go.dev/doc/effective_go

func main() {
	const Pi float64 = 3.14159
	var nombre string = "Go"
	var edad int = 30
	var esVerdadero bool = true
	var altura float64 = 1.75
	var b byte = 0x41
	var letra rune = 'ñ'
	var puntero *int = &edad
	var g map[string]int = map[string]int{"uno": 1, "dos": 2}
	type MiInterface interface{ MiMetodo() string }
	var f []int = []int{1, 2, 3}
	type H struct{ Campo string }
	fmt.Println("!Hola, " + nombre)
	imprimirDatos := fmt.Sprintf("Entero: %d\nCadena de Texto: %s\nBooleano: %t\nFlotante: %f\nByte: %c\nRuna: %c\nPuntero: %p\nMapa: %v\nArray: %v", edad, nombre, esVerdadero, altura, b, letra, puntero, g, f)
	fmt.Println(imprimirDatos)
}

File: Roadmap/00 - SINTAXIS, VARIABLES, TIPOS DE DATOS Y HOLA MUNDO/c/N0HagoNada.c
#include <stdio.h>
#include <stdbool.h>
#include <stdint.h> // Para tipos de tamaño fijo

// Una pagina oficial podria ser https://www.learn-c.org/
/*
Comentarios en varias lienas
https://www.gnu.org/software/gnu-c-manual/gnu-c-manual.html
*/

int main()
{
    // Creación de una variable
    int ejemploVariable = 10;

    // Creación de una constante
    const float PI = 3.1416;

    // Variables representando todos los tipos de datos primitivos
    char unCaracter = 'A'; // Carácter
    int unEntero = 123;    // Número entero
    unsigned int enteroSinSigno = 20;
    short corto = 10;                                         // Entero corto
    unsigned short cortoSinSigno = 20;                        // Entero corto sin signo
    long largo = 123456789;                                   // Entero largo
    unsigned long largoSinSigno = 987654321;                  // Entero largo sin signo
    long long largoLargo = 123456789101112;                   // Entero largo largo
    unsigned long long largoLargoSinSigno = 1098765432112345; //
    long long int entero64 = 1234567890123456789;
    float unFlotante = 123.456;                        // Número flotante
    double unDoble = 123.456789;                       // Número de doble precisión
    long double dobleLargo = 3.14159265358979323846L;  // Número de doble precisión largo
    uint64_t entero64SinSigno = 18446744073709551615U; // Entero sin signo de 64 bits
    bool unBooleano = true;                            // Booleano (requiere #include <stdbool.h>)
    char *unaCadena = "Esto es una cadena de texto";   // Cadena de texto

    // Imprime por terminal el texto solicitado
    printf("¡Hola, C!\n");

    // Imprime los valores de las variables (opcional, para demostración)
    printf("Variable int: %d\n", ejemploVariable);
    printf("Constante float: %f\n", PI);
    printf("Entero sin signo: %u\n", enteroSinSigno);
    printf("Corto: %hd\n", corto);
    printf("Corto sin signo: %hu\n", cortoSinSigno);
    printf("Largo: %ld\n", largo);
    printf("Largo sin signo: %lu\n", largoSinSigno);
    printf("Largo largo: %lld\n", largoLargo);
    printf("Largo largo sin signo: %llu\n", largoLargoSinSigno);
    printf("char: %c\n", unCaracter);
    printf("int: %d\n", unEntero);
    printf("float: %f\n", unFlotante);
    printf("double: %lf\n", unDoble);
    printf("Doble largo: %Lf\n", dobleLargo);
    printf("Entero 64 bits: %ld\n", entero64);
    printf("Entero 64 bits sin signo: %lu\n", entero64SinSigno);
    printf("bool: %d\n", unBooleano); // En C, los booleanos se imprimen como enteros (0 para false, 1 para true)
    printf("char*: %s\n", unaCadena);

    return 0;
}